From 7d341297377057829b588873d30bfdb6243ee315 Mon Sep 17 00:00:00 2001
From: Max Wipfli <max.wipfli@axelera.ai>
Date: Fri, 5 Jul 2024 07:44:34 +0000
Subject: [PATCH 6/9] Generate instructions.hex using global decoder APB base
 address

This change should not affect functionality, and is required for
integration into our address map.
---
 bin/mcu/src/mcu_hex2c.pl | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/bin/mcu/src/mcu_hex2c.pl b/bin/mcu/src/mcu_hex2c.pl
index e861135..cd2ad1f 100755
--- a/bin/mcu/src/mcu_hex2c.pl
+++ b/bin/mcu/src/mcu_hex2c.pl
@@ -64,6 +64,10 @@ close(OUT);
 
 ################
 # instructions.hex
+
+# These instructions will be executed by the APU, so need to use the decoder APB
+# base from the system POV (DECODER_BASE).
+$apbid=0xB0000000;
 open(OUT, '>instructions.hex') or die($!);
 print OUT "start_frame,0,MCU,0,                      // LOAD MCU CODE\n";
 print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x0014)).",0x40000000, // ENABLE IP IT\n";
-- 
2.43.0

